src/interface.ts

```diff
@@ -15,37 +23,39 @@ enum ResolveError {
   ResolverTimeout = "ResolverTimeout"
 }
 
-interface Intent {
-  intent: IntentName;
-  context: Context;
-  /**
-   * Name of app to target for the Intent. Use if creating an explicit intent
-   * that bypasses resolver and goes directly to an app.
-   */
-  target?: AppIdentifier;
-  
-  /**
-   * Dispatches the intent with the Desktop Agent.
-   * 
-   * Accepts context data and target (if an explicit Intent) as optional args.
-   * Returns a Promise - resolving if the intent successfully results in launching an App.
-   * If the resolution errors, it returns an `Error` with a string from the `ResolveError` enumeration.
-   */
-  send(context: Context, target?: AppIdentifier): Promise<void>
+/**
+ * Specified whether the target app is already running or it can be launched by invokation.
+ */
+enum AppState {
+  Launchable = 0,
+  Running = 1
 }
 
 /**
- * App metadata is Desktop Agent specific - but should support a name property.
+ * Specifies invokation target. Can be resolved from Agent or constructed directly.
+ * Caller can specify the concrete instance of app which should handle the invokation.
+ * Caller can specify if he want to route the invokation to already running or launchable instance.
  */
-interface AppMetadata {
-  name: AppIdentifier;
+interface Target {
+  intentName: IntentName,           
+  applicationName?: AppIdentifier,  // Optional. Can be undefined when constructed directly if target app is unknown and caller let Agent decide which app should handle the invokation.
+  instance?: AppInstanceIdentifier, // Optional. Can be undefined when constructed directly if target app instance is unknown and caller let Agent decide which instance should handle the invokation. If specified then Agent will route the context to this exact app instance.
+  state?: AppState                  // Optional. Can be undefined when constructed directly if target app state doesn't matter for caller. If state=Launchable then Agent will launch new instance of target app. If state=Running then Agent will route context to the already running instance.
 }
 
-interface Listener {
-  /**
-   * Unsubscribe the listener object.
-   */
-  unsubscribe();
+/**
+ * Allows cancelling asynchronous operation.
+ */
+interface Cancellable {
```

Comment:
I don't agree with this renaming. In the current code the `Listener` interface is used to unsubscribe from `contextListener` events. Cancellation implies that a given operation was reverted / cancelled, which isn't quite the same thing.